File: src/main/org/ascore/lang/modules/EnumModule.java
package org.ascore.lang.modules;

import org.ascore.lang.modules.builtins.ModuleBuiltin;
import org.ascore.lang.modules.core.ModuleFactory;
import org.ascore.lang.modules.core.ModuleManager;

public enum EnumModule {
    builtins(ModuleBuiltin::load),
    ;

    EnumModule(ModuleFactory moduleFactory) {
        ModuleManager.enregistrerModule(this, moduleFactory);
    }
}

File: src/org/ascore/lang/modules/corekotlin/EnumModule.kt
package org.ascore.lang.modules.corekotlin

import org.ascore.lang.modules.EnumModule
import org.ascore.lang.modules.core.ASModuleFactory
import org.ascore.lang.modules.core.ASModuleManager

enum class EnumModule(moduleFactory: ASModuleFactory?) {
    ;

    init {
        ASModuleManager.enregistrerModule(EnumModule.valueOf(this.toString()), moduleFactory)
    }
}